src/backend/api/mirrors_update.py

```diff
@@ -249,17 +249,19 @@ async def mirror_available(
                 async with http_session.get(
                     check_url,
                     headers=HEADERS,
-                    timeout=45,
+                    timeout=30,
                 ) as resp:
                     await resp.text()
-                    if resp.status == 200:
-                        # if mirror has at least one valid version/arch combo
-                        logger.info(
-                            'Mirror "%s" is available',
-                            mirror_name,
+                    if resp.status != 200:
+                        # if mirror has no valid version/arch combos it is dead
+                        logger.error(
+                            'Mirror "%s" has one or more invalid repositories',
+                            mirror_name
                         )
-                        return mirror_name, True
-            except (ClientError, asyncio.TimeoutError) as err:
+                        return mirror_name, False
+            except (ClientError, asyncio.exceptions.TimeoutError) as err:
+                if type(err) == asyncio.exceptions.TimeoutError:
```

Comment:
1. Please, use `isinstance(err, asyncio.exception.TimeoutError)`
2. And import `from asyncio.exceptions import TimeoutError` is better because it allows to use shorter the line here

Reply:
Fix.  I tend to prefer longer method usage versus more specific imports for clarity when looking at code in the future, but I changed this as you wanted.